### PR TITLE
Fix download links

### PR DESCRIPTION
**Pull request by bbodenmiller**
#### Summary

- Fix download links 
- Put them in consistent location for each release

### CODE REVIEW

File: source/install/desktop-app-changelog.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ Release v5.4
 
 **Release Day: June 19, 2023**
 
-**Download Binaries:** `Mattermost Desktop on GitHub <https://github.com/mattermost/desktop/releases/latest>`_
```
Comment (amyblais) [inline source/install/desktop-app-changelog.rst:21]: I don't have context on why we want to remove the "latest" link. Other changes in the PR look good to me.

Comment (bbodenmiller) [reply source/install/desktop-app-changelog.rst:21]: @amyblais might be okay for latest release but seems like historically folks have forgot to update this link when next release comes out so less error prone to point at release directly.

Comment (amyblais) [reply source/install/desktop-app-changelog.rst:21]: Makes sense, I think we can leave the "latest" link here, and I'll add to the release checklist to update the link when a new release is out.

Comment (amyblais) [reply source/install/desktop-app-changelog.rst:21]: If the submitter is ok with it, we can leave the "latest" link here, and I'll add to the release checklist to update the link when a new release is out.

Comment (amyblais) [reply source/install/desktop-app-changelog.rst:21]: If the submitter is ok with it, we can leave the "latest" link here, and I'll add to the release checklist to update the link when a new release is out.

File: source/install/desktop-app-changelog.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ Release v5.4
...
+**Download Binaries:** `Mattermost Desktop on GitHub <https://github.com/mattermost/desktop/releases/tag/v5.4.0>`_
```
Comment (bbodenmiller) [inline source/install/desktop-app-changelog.rst:21]:
```suggestion
**Download Binaries:** `Mattermost Desktop on GitHub <https://github.com/mattermost/desktop/releases/latest>`_
```

File: source/install/desktop-app-changelog.rst (hunk 1 above)
Comment (bbodenmiller) [reply source/install/desktop-app-changelog.rst]: @amyblais sounds good updated

**Review by amyblais — APPROVED** [review]

**Comment by amyblais** [comment]
Thank you!

**Pull request merged**